check_logs.py

```diff
@@ -13,7 +14,36 @@
 def _fetch(title, url, dest):
     print_yellow("fetching %s from: %s" % (title, url))
     # TODO: use something more robust like python wget library.
-    urllib.request.urlretrieve(url, dest)
+    retries = 0
+    max_retries = 7
+    retry_codes = [500, 504]
+    while retries < max_retries:
+        try:
+            time.sleep(2**retries)
+            urllib.request.urlretrieve(url, dest)
+            retries += 1
+            break
+        except ConnectionResetError:
+            pass
+        except HTTPError as err:
+            if err.code in retry_codes:
+                pass
+            elif err.code == 404:
+                print_red(
+                    "%s could not be found (404 error), did the build timeout?"
+                    % (title))
+                sys.exit(1)
+            else:
+                print_red("%d error trying to download %s" % (err.code, title))
+                sys.exit(1)
+        except URLError:
+            pass
+
```

Comment:
I think it might be nice to include details about each failure case. Right now, this will silently retry with no logging of which of the conditional failures were encountered. i.e. perhaps something like this before each "pass" case:

```
   print_yellow("%s download failure: %s" % (str(err)))
   pass
```